### PR TITLE
Add mainnet solver rewards dashboard and update with service fee

### PR DESCRIPTION
**Pull request by harisang**
This PR adds the following to the repo:
- [Query 2510345](https://dune.com/queries/2510345), which is the main query showing in the solver rewards [dashboard](https://dune.com/cowprotocol/cow-solver-rewards?StartTime=2024-08-20+00%3A00%3A00&EndTime=2024-08-27+00%3A00%3A00).
- [Query 4017925](https://dune.com/queries/4017925), which computes whether a solver needs to pay the CoW DAO bonding pool service fee or not.

Note that query 2510345 has some proposed changes, regarding the implementation of the service fee and the reduction in the rewards being transferred to solvers. The changes are between lines 746-811.

So the plan is to get the PR approved, and before merging actually edit query 2510345 to ensure its code is identical to the code in this PR.

### CODE REVIEW

File: mainnet_solver_rewards/service_fee_query_4017925.sql [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+),
+
+named_results AS (
+  SELECT
+    jd.solver,
+    CONCAT(environment, '-', s.name) AS solver_name,
+    jd.pool_name,
+    jd.pool,
+    jd.joined_on,
+    date_diff('day', date(jd.joined_on), date(NOW())) AS days_in_pool
```
Comment (fhenneke) [inline mainnet_solver_rewards/service_fee_query_4017925.sql:117]: `NOW()` should probably be changed to a `cutoff_time` parameter which would chosen as `start_time`.

File: mainnet_solver_rewards/service_fee_query_4017925.sql [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    VALUES {{bondingpooldata}}
+  ) AS _ (pool, name, funder)
+),
+
+first_event_after_timestamp AS (
+  SELECT MAX(number) FROM ethereum.blocks
+  WHERE time > CAST('2024-08-20 00:00:00' AS timestamp) -- CIP-48 starts bonding pool timer at midnight UTC on 20/08/24
+),
+
+initial_vouches AS (
```
Comment (fhenneke) [inline mainnet_solver_rewards/service_fee_query_4017925.sql:13]: The part on vouches might just reuse the vouches query.

File: mainnet_solver_rewards/service_fee_query_4017925.sql
Comment (fhenneke) [inline mainnet_solver_rewards/service_fee_query_4017925.sql:1]: That query does not look much simpler that just hard-coding all solver addresses, their starting date, and their status (e.g. exempt for 6 month, 9 month, indefinitely).

The query might help finding those dates.

File: mainnet_solver_rewards/service_fee_query_4017925.sql [hunk 3]
```diff
@@ -0,0 +1,175 @@
+WITH
+bonding_pools (pool, name, initial_funder) AS (
+  SELECT from_hex(pool), name, from_hex(funder) FROM (
```
Comment (fhenneke) [inline mainnet_solver_rewards/service_fee_query_4017925.sql:3]: (this query uses a different style (keywords in all caps) than the other query)

**Review by fhenneke — CHANGES_REQUESTED** [review]
This is a good opportunity to clean up this query (and the respective solver-rewards query).

As a minimum, the external queries referenced in this query should be called instead of their code being copied.

There is much room for other improvements, but some of that (e.g. using the actual solver-rewards query for the dashboard or using query 4021306 from coincidence of want for slippage) could be addressed later.

**Review by fhenneke — APPROVED** [review]

**Comment by harisang** [comment]
Closing as a lot of changes have happened in between so will reopen a PR with just the main dashboard there.

**Pull request closed without merging**